fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateRoutineLoadInfo.java

```diff
@@ -0,0 +1,525 @@
+
+    /**
+     * constructor for create table
+     */
+    public CreateRoutineLoadInfo(LabelName labelName, String tableName, Map<String, LoadProperty> loadPropertyMap,
+                                 Map<String, String> jobProperties, String typeName,
+                                 Map<String, String> dataSourceProperties, LoadTask.MergeType mergeType,
+                                 String comment) {
+        this.labelName = labelName;
+        if (StringUtils.isBlank(tableName)) {
+            this.isMultiTable = true;
+        }
+        this.tableName = tableName;
+        this.loadPropertyMap = loadPropertyMap;
+        this.jobProperties = jobProperties == null ? Maps.newHashMap() : jobProperties;
+        this.typeName = typeName.toUpperCase();
+        this.dataSourceProperties = RoutineLoadDataSourcePropertyFactory
+            .createDataSource(typeName, dataSourceProperties, this.isMultiTable);
+        this.mergeType = mergeType;
+        this.isPartialUpdate = this.jobProperties.getOrDefault(PARTIAL_COLUMNS, "false").equalsIgnoreCase("true");
+        if (comment != null) {
+            this.comment = comment;
+        }
+    }
+
+    /**
+     * analyze create table info
+     */
+    public void validate(ConnectContext ctx) throws UserException {
+        // check dbName and tableName
+        checkDBTable(ctx);
+        // check name
+        try {
+            FeNameFormat.checkCommonName(NAME_TYPE, name);
+        } catch (org.apache.doris.common.AnalysisException e) {
+            // 64 is the length of regular expression matching
+            // (FeNameFormat.COMMON_NAME_REGEX/UNDERSCORE_COMMON_NAME_REGEX)
+            throw new AnalysisException(e.getMessage()
+                + " Maybe routine load job name is longer than 64 or contains illegal characters");
+        }
+        // check load properties include column separator etc.
+        checkLoadProperties(ctx);
+        // check routine load job properties include desired concurrent number etc.
+        checkJobProperties();
+        // check data source properties
+        checkDataSourceProperties();
+        // analyze merge type
+        if (routineLoadDesc != null) {
+            if (mergeType != LoadTask.MergeType.MERGE && routineLoadDesc.getDeleteCondition() != null) {
+                throw new AnalysisException("not support DELETE ON clause when merge type is not MERGE.");
+            }
+            if (mergeType == LoadTask.MergeType.MERGE && routineLoadDesc.getDeleteCondition() == null) {
+                throw new AnalysisException("Excepted DELETE ON clause when merge type is MERGE.");
+            }
+        } else if (mergeType == LoadTask.MergeType.MERGE) {
+            throw new AnalysisException("Excepted DELETE ON clause when merge type is MERGE.");
+        }
+    }
+
+    private void checkDBTable(ConnectContext ctx) throws AnalysisException {
+        dbName = labelName.getDbName();
```

Comment:
Better to create a new LabelNameInfo class and implement analyze or validate method like TableNameInfo. Because LabelNameInfo may be used in other commands, and the validate can be called in multiple place.

Reply:
done